tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -136,4 +184,18 @@ object Build {
     }
   }
 
+  private def getLastModifiedTimeMillis(path: Path): Long =
+    if (Files.exists(path)) Files.getLastModifiedTime(path).toMillis()
+    else 0L
+
+  /** Get the newest file's last modified time in millis under the given path.
+   */
+  private def getNewestMtime(path: Path): Optional[FileTime] =
+    if (Files.exists(path))
+      Files
+        .walk(path, FileVisitOption.FOLLOW_LINKS)
+        .map[FileTime](Files.getLastModifiedTime(_))
```

Comment:
It's possible that this call might throw an exception if the file was removed in the meantime or we don't have permissions. I think it would be best to refer to the getLastModifiedTimeMillis defined above (possibly both can reduce FileTime (and FileTime.fromMilllis(0)) for non existing files)

Reply:
Right, I made a change like this https://github.com/scala-native/scala-native/pull/3287/commits/5a4ee264a5b5d966fb2600c25ce5c44f5feda537

Maybe we should print warning for Exceptions in `getLastModified`?